### PR TITLE
Fix remaining `STRICT` compiler warnings

### PR DESCRIPTION
**Pull request by mikee47**
Core code (excluding Libraries) should build entirely without warnings or errors.

### CODE REVIEW

File: Sming/SmingCore/DateTime.cpp [hunk 1]
```diff
@@ -379,7 +379,7 @@ String DateTime::format(String sFormat)
 void DateTime::calcDayOfYear()
 {
 	DayofYear = 0;
-	for(unsigned int i = 0; i < Month; ++i) {
+	for(int i = 0; i < Month; ++i) {
```
Comment (slaff) [inline Sming/SmingCore/DateTime.cpp]: Just for my understanding: Why not use `unsigned` here?

File: Sming/SmingCore/Network/TcpServer.cpp [hunk 2]
```diff
@@ -129,7 +129,8 @@ err_t TcpServer::onAccept(tcp_pcb* clientTcp, err_t err)
 	connections.add(client);
 	debug_d("Opening connection. Total connections: %d", connections.count());
 
-	onClient((TcpClient*)client);
+	// @todo Dangerous up-cast - needs looking at
```
Comment (slaff) [inline Sming/SmingCore/Network/TcpServer.cpp]: I guess you will take a look at this before the PR is declared ready for merging?

File: Sming/SmingCore/DateTime.cpp (hunk 1 above)
Comment (mikee47) [reply Sming/SmingCore/DateTime.cpp]: `Month` is signed (`int8_t`) as are all the member variables and methods. I can re-write the `DateTime` class using unsigned values, which is rather more extensive and would be a separate PR.
This seemed like an easier solution unless there's actually a problem with `DateTime`. What'cha wanna do?

File: Sming/SmingCore/Network/TcpServer.cpp (hunk 2 above)
Comment (mikee47) [reply Sming/SmingCore/Network/TcpServer.cpp]: Here be demons. We have `virtual createClient()` returning a `TcpConnection*` which, in most cases, is actually a `TcpClient*`, except for `FTPServerConnection` which creates an `FTPServerConnection` inherited directly from `TcpConnection`. Logically, I'd expect `createClient` to `TcpClient*'. If that's the case it's a significant change to the framework and FTPServer needs looking at - there's a lot of extra stuff in `TcpClient` vs. `TcpConnection`.

If you have any thoughts on this, most welcome!

This'll need a separate PR, so I'll remove the comment.

File: Sming/SmingCore/DateTime.cpp (hunk 1 above)
Comment (slaff) [reply Sming/SmingCore/DateTime.cpp]: > This seemed like an easier solution unless there's actually a problem with `DateTime`

Then it is ok.

File: Sming/SmingCore/Network/TcpServer.cpp (hunk 2 above)
Comment (slaff) [reply Sming/SmingCore/Network/TcpServer.cpp]: > This'll need a separate PR, so I'll remove the comment and put up with the 'C-style typecast' warning for now.

Perfect. Thanks!

Comment (slaff) [reply Sming/SmingCore/Network/TcpServer.cpp]: Just remove the comment so that I can merge the PR.

File: Sming/SmingCore/DateTime.cpp (hunk 1 above)
Comment (mikee47) [reply Sming/SmingCore/DateTime.cpp]: > unless there's actually a problem with `DateTime`

Had a read through and found a few things. Will submit separate PR for this.

File: Sming/SmingCore/Network/TcpServer.cpp (hunk 2 above)
Comment (slaff) [reply Sming/SmingCore/Network/TcpServer.cpp]: @mikee47 Is the PR ready for merging?

Comment (mikee47) [reply Sming/SmingCore/Network/TcpServer.cpp]: @slaff Yes, all done.

**Pull request merged**